File: 001_binary_exponentiation/implementations/php/main.php
<?php

function binExp(int $a, int $b): int {
    $result = 1;
    while($b > 0) {
        if ($b % 2 != 0) $result *= $a;
        $a *= $a;
        $b = floor($b/2);
    }
    return $result;
}

echo binExp(3, 9);

File: 001_binary_exponentiation/implementations/dart/main.dart
int binExp(int a, int b) {
  var result = 1;
  while (b > 0) {
    if (b % 2 != 0) result *= a;
    a *= a;
    b ~/= 2;
  }
  return result;
}

void main() {
  print(binExp(3, 22));
}
